provider/json/watcher.go

```diff
@@ -0,0 +1,109 @@
+package json
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"io"
+	"strings"
+
+	"github.com/src-d/lookout"
+
+	"gopkg.in/src-d/go-log.v1"
+)
+
+// Provider is the name
+const Provider = "json"
+
+// WatchOptions options to use in the Watcher constructor.
+type WatchOptions struct {
```

Comment:
Could you please explain me the reason to have this struct?

Github provider accepts `lookout.WatchOptions` which should be common for different providers.
I see currently there is only `URL` prop which doesn't make any sense for console provider.

I would suggest to remove this struct to avoid confusion and change constructor to either:
- `NewWatcher(io.Reader)`
- `NewWatcher(io.Reader, lookout.WatchOptions)`

Reply:
As you said, `lookout.WatchOptions` should be common, but the current `URL` does not apply to this new provider. I just put `WatchOptions` to make it consistent.

If we change this to `NewWatcher(io.Reader)`, do you think we should remove `lookout.WatchOptions` and make `github NewWatcher(transport http.RoundTripper, URL string)`?

Reply:
to make it consistent I would do:
```go
func NewWatcher(io.Reader, *lookout.WatchOptions)
```

similar to github
```go
func NewWatcher(http.RoundTripper, *lookout.WatchOptions)
```

The first argument is "the source" (http for github and io for console), the second is common options.

About removing `lookout.WatchOptions` completely:

I would keep it at least for now. I see how it can be useful for other providers (gitlab, bitbucket, ...). Console provider is a way too special.